File: python/reverse_list.py
def reverse(n:list) -> list:
    if len(n) <= 2:
        raise ValueError("invalid array")
    
    return n[::-1]

def main():
    arr = [1,2,3,4]

    reverseArr = reverse(arr)
    print(f"Reverse array is {reverseArr}")

main()    

File: c/reverse_list.c
#include <stdio.h>
#include <stdlib.h>

void reverse(int* arr, int size);

int main() {
    int arr[] = {1,2,4};
    int size = sizeof(arr) / sizeof(arr[0]);

    reverse(arr, size);

    for(int i = 0; i < size; i++) {
        printf("%i ", arr[i]);
    }
    printf("\n");

    return EXIT_SUCCESS;
}

void reverse(int* arr, int size) {

    if(size <= 1){
        return;
    }

    printf("passsing\n");
    int temp;
    for(int i = 0; i < size / 2; i++) {
        // swap arr[i] with arr[size - 1 - i]
        temp = arr[i];
        arr[i] = arr[size - 1 - i];
        arr[size - 1 - i] = temp;
    }
}
